### PR TITLE
Release 2.3.0-beta.1

### PR DESCRIPTION
**Pull request by rreusser**
GL-JS and style-spec changelog and version number updates for release-2.3.0-beta.1.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
+## 2.3.0-beta.1
+
+### ✨ Features and improvements
+* Add configurable fog as a root style specification ([#10564](https://github.com/mapbox/mapbox-gl-js/pull/10564))
+* Add `map.queryTerrainElevation(lngLat)` API ([#10602](https://github.com/mapbox/mapbox-gl-js/pull/10602))
+* Reuse one renderbuffer for all draping. ([#10611](https://github.com/mapbox/mapbox-gl-js/pull/10611))
```
Comment (karimnaaji) [inline CHANGELOG.md:6]:
```suggestion
* Reduce GPU memory footprint when terrain is active by sharing a single depth stencil renderbuffer for all draping ([#10611](https://github.com/mapbox/mapbox-gl-js/pull/10611))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
+## 2.3.0-beta.1
+
+### ✨ Features and improvements
+* Add configurable fog as a root style specification ([#10564](https://github.com/mapbox/mapbox-gl-js/pull/10564))
+* Add `map.queryTerrainElevation(lngLat)` API ([#10602](https://github.com/mapbox/mapbox-gl-js/pull/10602))
```
Comment (karimnaaji) [inline CHANGELOG.md:5]:
```suggestion
* Add client-side elevation querying with `map.queryTerrainElevation(lngLat)` when terrain is active ([#10602](https://github.com/mapbox/mapbox-gl-js/pull/10602))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+* Reuse one renderbuffer for all draping. ([#10611](https://github.com/mapbox/mapbox-gl-js/pull/10611))
+* Add support for data-driven `text-line-height` ([#10612](https://github.com/mapbox/mapbox-gl-js/pull/10612))
+* Add support for data-driven expressions in `line-dasharray` and `line-cap` properties. ([#10591](https://github.com/mapbox/mapbox-gl-js/pull/10591))
+* Prevent unnecessary loading of tiles when terrain is active ([#10467](https://github.com/mapbox/mapbox-gl-js/pull/10467))
```
Comment (karimnaaji) [inline CHANGELOG.md:9]:
```suggestion
Optimize tile cover by preventing unnecessary tile loads when terrain is active ([#10467](https://github.com/mapbox/mapbox-gl-js/pull/10467))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+
+### 🐞 Bug fixes
+* Fix style property transitions not invalidating the terrain render cache ([#10485](https://github.com/mapbox/mapbox-gl-js/pull/10485))
+* Fix raster tile expiry data not being retained. ([#10494](https://github.com/mapbox/mapbox-gl-js/pull/10494)) (h/t [andycalder](https://github.com/andycalder))
+* Fix unclustered points in a clustered GeoJSON source incorrectly snapping to a grid at high zoom levels. ([#10523](https://github.com/mapbox/mapbox-gl-js/pull/10523))
+* Fix `map.loadImage` followed by `map.addImage` with a delay failing in Safari and Firefox. ([#10524](https://github.com/mapbox/mapbox-gl-js/pull/10524))
+* Allow formatted images in text to be displayed conditionally ([#10553](https://github.com/mapbox/mapbox-gl-js/pull/10553))
+* Fix fill extrusion elevation underflow below sea level ([#10570](https://github.com/mapbox/mapbox-gl-js/pull/10570))
```
Comment (karimnaaji) [inline CHANGELOG.md:17]:
```suggestion
* Fix fill-extrusion elevation underflow below sea level ([#10570](https://github.com/mapbox/mapbox-gl-js/pull/10570))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+* Fix dashed lines with square line caps. ([#9561](https://github.com/mapbox/mapbox-gl-js/pull/9561))
+* Batch render DOM elements to avoid reflow ([#10530](https://github.com/mapbox/mapbox-gl-js/pull/10530), [#10567](https://github.com/mapbox/mapbox-gl-js/pull/10567)) (h/t [zarov](https://github.com/zarov))
```
Comment (karimnaaji) [inline CHANGELOG.md:19]: This one is a bit on the line between bugfix and optimization, we could also mark it as an optimization for the improvements section. But also fine as is.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+* Fix markers sometimes throwing an error after being removed from a 3D map. ([#10478](https://github.com/mapbox/mapbox-gl-js/pull/10478)) (h/t [andycalder](https://github.com/andycalder))
+* Set attribution button type=button to prevent form submit when map nested in form ([#10531](https://github.com/mapbox/mapbox-gl-js/pull/10531))
+* Fix nine documentation typos ([#10546](https://github.com/mapbox/mapbox-gl-js/pull/10546), [#10548](https://github.com/mapbox/mapbox-gl-js/pull/10548) [#10551](https://github.com/mapbox/mapbox-gl-js/pull/10551) [#10646](https://github.com/mapbox/mapbox-gl-js/pull/10646)) (h/t [coliff](https://github.com/coliff)
+* Fix undefined type error when removing `line-gradient` paint property ([#10557](https://github.com/mapbox/mapbox-gl-js/pull/10557))
```
Comment (karimnaaji) [inline CHANGELOG.md:23]: I think this bugfix is a bit more important than the last two and should be a bit higher up in the list.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,27 @@
...
+* Fix fill extrusion elevation underflow below sea level ([#10570](https://github.com/mapbox/mapbox-gl-js/pull/10570))
+* Fix dashed lines with square line caps. ([#9561](https://github.com/mapbox/mapbox-gl-js/pull/9561))
+* Batch render DOM elements to avoid reflow ([#10530](https://github.com/mapbox/mapbox-gl-js/pull/10530), [#10567](https://github.com/mapbox/mapbox-gl-js/pull/10567)) (h/t [zarov](https://github.com/zarov))
+* Fix markers sometimes throwing an error after being removed from a 3D map. ([#10478](https://github.com/mapbox/mapbox-gl-js/pull/10478)) (h/t [andycalder](https://github.com/andycalder))
+* Set attribution button type=button to prevent form submit when map nested in form ([#10531](https://github.com/mapbox/mapbox-gl-js/pull/10531))
+* Fix nine documentation typos ([#10546](https://github.com/mapbox/mapbox-gl-js/pull/10546), [#10548](https://github.com/mapbox/mapbox-gl-js/pull/10548) [#10551](https://github.com/mapbox/mapbox-gl-js/pull/10551) [#10646](https://github.com/mapbox/mapbox-gl-js/pull/10646)) (h/t [coliff](https://github.com/coliff)
```
Comment (karimnaaji) [inline CHANGELOG.md:22]: I'm not against removing that entry. We have a lot of things in that changelog and this one isn't _that_ important (unless we find it important to h/t the external contributions there).

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,27 @@
...
+* Fix `map.loadImage` followed by `map.addImage` with a delay failing in Safari and Firefox. ([#10524](https://github.com/mapbox/mapbox-gl-js/pull/10524))
+* Allow formatted images in text to be displayed conditionally ([#10553](https://github.com/mapbox/mapbox-gl-js/pull/10553))
+* Fix fill extrusion elevation underflow below sea level ([#10570](https://github.com/mapbox/mapbox-gl-js/pull/10570))
+* Fix dashed lines with square line caps. ([#9561](https://github.com/mapbox/mapbox-gl-js/pull/9561))
+* Batch render DOM elements to avoid reflow ([#10530](https://github.com/mapbox/mapbox-gl-js/pull/10530), [#10567](https://github.com/mapbox/mapbox-gl-js/pull/10567)) (h/t [zarov](https://github.com/zarov))
+* Fix markers sometimes throwing an error after being removed from a 3D map. ([#10478](https://github.com/mapbox/mapbox-gl-js/pull/10478)) (h/t [andycalder](https://github.com/andycalder))
+* Set attribution button type=button to prevent form submit when map nested in form ([#10531](https://github.com/mapbox/mapbox-gl-js/pull/10531))
```
Comment (karimnaaji) [inline CHANGELOG.md:21]:
```suggestion
* Set attribution button `type=button` to prevent form submit when map nested in form ([#10531](https://github.com/mapbox/mapbox-gl-js/pull/10531))
```

File: CHANGELOG.md (hunk 1 above)
Comment (karimnaaji) [reply CHANGELOG.md:6]: A bit wordy so feel free to rephrase, but just want to explicitly highlight what this change does for the user (e.g. _Reducing GPU memory footprint_)

File: CHANGELOG.md (hunk 2 above)
Comment (karimnaaji) [reply CHANGELOG.md:5]: Feel free to rephrase this one, but I find it important to highlight two things:
- Available when terrain is on
- Prevents you from doing any sort of server query

File: CHANGELOG.md (hunk 6 above)
Comment (rreusser) [reply CHANGELOG.md:23]: Sounds good. It wasn't clear to me looking back at previous changelog entries exactly how the ordering was decided, so I mostly didn't think too much about it.

File: CHANGELOG.md (hunk 7 above)
Comment (rreusser) [reply CHANGELOG.md:22]: I believe it said h/t entries no matter how small 🙏 🙇

File: CHANGELOG.md (hunk 5 above)
Comment (rreusser) [reply CHANGELOG.md:20]: I'll promote it to feature. I contemplated since it was phrased specifically as "a bugfix of #_____", but agreed, I think it's more of a feature.

**Review by karimnaaji — APPROVED** [review]
